doc/UnifiedTeamD/Unified_TeamD_HLD.md

```diff
@@ -0,0 +1,426 @@
+
+**IPC message for Portchannel Member Deletion**
+
+```
+| REQUEST  | PortRemove  | <portchannel_name> | <port_member>
+  <opcode>   <msg_type> 
+```
+
+In addition to these, the existing IPC message formats for port member addition and removal are reused. The key change is that these messages are now initiated by teammgrd instead of teamd_ctl calls. Furthermore, each message now includes the name of the associated port channel.
+
+<div align="center"> <img src=images/portchannel_member_add_del.png width=600 /> </div>
+
+### State DB Update Flow for LAG and Member
+
+When a port goes UP/Down, the Linux kernel notifies teamsyncd via netlink. Teamsyncd processes the event and updates the relevant LAG entries in the STATE_DB and APP_DB , reflecting the operational status. The tlm_teamd component, which has subscribed to the LAG_TABLE in STATE_DB, gets notified of the update. Additionally, tlm_teamd sends a state dump request to teamd over a UNIX domain socket (IPC), prompting teamd to return its current view of LAG states. This flow ensures end-to-end synchronization of LAG status between the kernel, database layers, and the teamd process, maintaining accurate and consistent LAG state information throughout the system.
+
+**IPC Request**
+
+msg_type - StateDump
+
+```
+| REQUEST  | StateDump  | <portchannel_name> | <ipc_response>
+  <opcode>   <msg_type> 
+```
+
+**IPC Response**
+```
+REPLY_SUCCESS <response>
+```
+
+<div align="center"> <img src=images/link_state.png width=600 /> </div>
+
+### Link Event Flow
+When a link state change occurs, the kernel sends a netlink notification, which is received by the registered libteam callback. Libteam parses the netlink message, extracts the index of the affected link, and invokes a lookup function registered by teamd. Teamd then retrieves the corresponding team_dev structure from its internal hash table and returns it to libteam. Using this structure, libteam processes the state change and informs teamd through its change handlers. Teamd then triggers the appropriate LACP logic to handle the link state transition, and any updates to link aggregation are communicated back to the kernel via libteam.
+
+<div align="center"> <img src=images/link_event_flow.png width=600 /> </div>
+
+
+
+## teamd operating mode 
+
+teamd shall support both unified and multi-process (legacy) operation modes. By default teamd will operate in unified process mode. But if an user wants to revert to legacy teamd, then user has to update this mode into config DB.
+
+### DB Changes
+
+#### Config DB 
+A new table named TEAMD will be introduced in the Config DB to support both the legacy and the new design.
+
+```
+{
+  "TEAMD" : {
+    "GLOBAL" : {
```

Comment:
Keep the legacy mode as default until we're comfortable making unified mode as default.